beater/interceptors/logging.go

```diff
@@ -0,0 +1,68 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package interceptors
+
+import (
+	"context"
+	"net/http"
+	"time"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/peer"
+	"google.golang.org/grpc/status"
+
+	"github.com/elastic/apm-server/utility"
+	"github.com/elastic/beats/v7/libbeat/logp"
+)
+
+// Logging intercepts a gRPC request and provides logging processing. The
+// returned function implements grpc.UnaryServerInterceptor.
+func Logging(logger *logp.Logger) grpc.UnaryServerInterceptor {
```

Comment:
Even with a system test I think it would be good to have a unit test for this. The unit test can be a bit more comprehensive, checking source address, forwarded headers, errors, etc. The system test would be more about checking the interceptors are wired up.